core/trino-main/src/main/java/io/trino/operator/output/SkewedPartitionRebalancer.java

```diff
@@ -303,7 +303,15 @@ private void calculatePartitionDataSize(long dataProcessed)
         }
 
         for (int partition = 0; partition < partitionCount; partition++) {
-            partitionDataSize[partition] = (partitionRowCount.get(partition) * dataProcessed) / totalPartitionRowCount;
+            // Since we estimate the partitionDataSize based on partitionRowCount and total data processed. It is possible
+            // that the estimated partitionDataSize is slightly less than it was estimated at the last rebalance cycle.
+            // That's because for a given partition, row count hasn't increased, however overall data processed
+            // has increased. Therefore, we need to make sure that the estimated partitionDataSize should be
+            // at least partitionDataSizeAtLastRebalance. Otherwise, it will affect the ordering of minTaskBuckets
+            // priority queue.
+            partitionDataSize[partition] = max(
+                    (partitionRowCount.get(partition) * dataProcessed) / totalPartitionRowCount,
+                    partitionDataSizeAtLastRebalance[partition]);
```

Comment:
why not `partitionDataSizeAtLastRebalance[partition]` -> `partitionDataSize[partition]`?

t0 -> partitionDataSize[partition] = 100 (estimated)
t1 -> partitionDataSize[partition] = 200 (estimated)
t2 -> partitionDataSize[partition] = 190 (estimated) !!! keep 200

t0 -> partitionDataSize[partition] = 100 (estimated)
t1 -> partitionDataSize[partition] = 200 (estimated)
t2 -> partitionDataSize[partition] = 190 (estimated) !!!  partitionDataSizeAtLastRebalance = 100

```
            partitionDataSize[partition] = max(
                    (partitionRowCount.get(partition) * dataProcessed) / totalPartitionRowCount,
                    partitionDataSize[partition]);
```
never goes down!

```
            partitionDataSize[partition] = max(
                    (partitionRowCount.get(partition) * dataProcessed) / totalPartitionRowCount,
                    partitionDataSizeAtLastRebalance[partition]);
```
can go down? cannot go down?